v2/pkg/genruntime/secrets/secrets.go

```diff
@@ -124,12 +123,11 @@ func checkSecretOwner(obj client.Object, secret *v1.Secret) error {
 	}
 
 	if !owned {
-		return errors.Errorf(
-			"cannot overwrite secret %s/%s which is not owned by %s/%s",
-			secret.GetNamespace(),
-			secret.GetName(),
-			secret.GetNamespace(),
-			obj.GetName())
+		return &SecretNotOwnedError{
+			Namespace:  secret.GetNamespace(),
+			SecretName: secret.GetName(),
+			ObjectName: obj.GetName(),
+		}
```

Comment:
Should this be a factory method that takes just `secret` and `obj`, extracting the information desired?

Reply:
I've moved to use a `NewSecretNotOwnedError` method here, but still taking the 3 individual parameters. This is the only usage of the constructor though so this method is effectively the factory method you're talking about. I don't want to tightly couple the `SecretNotOwnedError` to always expecting those inputs though as in the future it's likely to be expanded to cover the KeyVault case too (whenever we implement that).